IBPSA/Fluid/HeatPumps/ModularReversible/BaseClasses/PartialReversibleRefrigerantMachine.mo

```diff
@@ -0,0 +1,633 @@
+    annotation (Dialog(
+      group="Heat Losses",
+      tab="Condenser",
+      enable=use_conCap));
+
+  parameter Modelica.Units.SI.Density rhoCon=MediumCon.density(staCon_nominal)
+    "Condenser medium density"
+    annotation (Dialog(tab="Condenser", group="Medium properties"));
+  parameter Modelica.Units.SI.SpecificHeatCapacity cpCon=
+      MediumCon.specificHeatCapacityCp(staCon_nominal)
+    "Condenser medium specific heat capacity"
+    annotation (Dialog(tab="Condenser", group="Medium properties"));
+
+  //Evaporator
+  parameter Modelica.Units.SI.Time tauEva=30
+    "Evaporator heat transfer time constant at nominal flow"
+    annotation (Dialog(tab="Evaporator", group="Dynamics"));
+  parameter Modelica.Units.SI.Temperature TEva_nominal
+    "Nominal temperature of evaporator medium"
+    annotation (Dialog(group="Nominal condition"));
+  parameter Modelica.Units.SI.TemperatureDifference dTEva_nominal
+    "Nominal temperature difference in evaporator medium"
+    annotation (Dialog(group="Nominal condition", tab="Evaporator"));
+  parameter Modelica.Units.SI.MassFlowRate mEva_flow_nominal
+    "Nominal mass flow rate of the evaporator medium"
+    annotation (Dialog(
+      group="Nominal condition",
+      tab="Evaporator"));
+
+  parameter Modelica.Units.SI.PressureDifference dpEva_nominal(displayUnit="Pa")
+    "Pressure drop at nominal mass flow rate" annotation (Dialog(group="Flow resistance",
+        tab="Evaporator"));
+  parameter Real deltaMEva=0.1
+    "Fraction of nominal mass flow rate where transition to turbulent occurs"
+    annotation (Dialog(tab="Evaporator", group="Flow resistance"));
+  parameter Boolean use_evaCap=true
+    "=true if using capacitor model for evaporator heat loss estimation"
+    annotation (Dialog(group="Heat Losses", tab="Evaporator"),
+                                          choices(checkBox=true));
+  parameter Modelica.Units.SI.HeatCapacity CEva
+    "Heat capacity of the evaporator"
+    annotation ( Dialog(
+      group="Heat Losses",
+      tab="Evaporator",
+      enable=use_evaCap));
+  parameter Modelica.Units.SI.ThermalConductance GEvaOut
+    "Outer thermal conductance for evaporator heat loss calculations"
+    annotation ( Dialog(
+      group="Heat Losses",
+      tab="Evaporator",
+      enable=use_evaCap));
+  parameter Modelica.Units.SI.ThermalConductance GEvaIns
+    "Inner thermal conductance for evaporator heat loss calculations"
+    annotation ( Dialog(
+      group="Heat Losses",
+      tab="Evaporator",
+      enable=use_evaCap));
+  parameter Modelica.Units.SI.Density rhoEva=MediumEva.density(staEva_nominal)
+    "Evaporator medium density"
+    annotation (Dialog(tab="Evaporator", group="Medium properties"));
+  parameter Modelica.Units.SI.SpecificHeatCapacity cpEva=
+      MediumEva.specificHeatCapacityCp(staEva_nominal)
+    "Evaporator medium specific heat capacity"
+    annotation (Dialog(tab="Evaporator", group="Medium properties"));
+
```

Comment:
Why not making these `final`? What is the use case in which a user wants to change them from the default assignments?

Reply:
See reason above